### PR TITLE
Update module to pass ansible-test sanity. Update integration tests.

### PR DESCRIPTION
**Pull request by vbotka**
Hi! Would it be possible to review the updated module and tests, and accept this PR? Would you like to submit it to the community.general? I've already tested the module in my [fork](https://github.com/vbotka/community.general/). The ansible-test sanity [passed](https://gist.github.com/vbotka/978a79d6be4f5126cc969b0aa849082d). And the updated iocage_test.yml passed as well

```
PLAY RECAP ***********************************************************************************
host1 : ok=64   changed=10   unreachable=0    failed=0    skipped=0    rescued=5    ignored=0
```
I've documented some of the changes in the included CHANGELOG, updated .gitignore, and added both .ansible-lint and .yamllint.

My intention is to use the Ansible module [ucl](https://github.com/vbotka/ansible-ucl) to configure iocage. I'm looking forward to your comments.

Thank you,

Vladimir Botka

### CODE REVIEW

File: iocage_test.yml
Comment (xmj) [inline iocage_test.yml:2]: consider keeping `localhost` here, it'll be a lot easier for most of us to test it without a host/VM called `test_iocage`

File: .ansible-lint
Comment (xmj) [inline .ansible-lint:9]: Please leave a note why you commented this out, or remove it.

**Review by xmj — CHANGES_REQUESTED** [review]
Good start, please consider the following:

- keep license as-is, you won't get approval for GPL.
- it might make sense to use f-strings everywhere

File: CHANGELOG
Comment (vbotka) [reply CHANGELOG:24]: OK.

File: CHANGELOG
Comment (vbotka) [reply CHANGELOG:1]: Done.

File: iocage.py
Comment (vbotka) [reply iocage.py:4]: Done.

File: iocage.py
Comment (vbotka) [reply iocage.py:108]: I've added all of them. Make your choice.

File: iocage.py
Comment (vbotka) [reply iocage.py:4]: Done. Thank you.

File: iocage.py
Comment (vbotka) [reply iocage.py:324]: Yes, this is very helpful.

**Review by xmj — CHANGES_REQUESTED** [review]
Hi, with f-strings I meant the literal `f"string {variable1} {variable2}"` feature

Check e.g. 
https://medium.com/@bvvkrishna/f-strings-python-3-8-string-formatting-d69b9c4e47a

File: .ansible-lint
Comment (vbotka) [reply .ansible-lint]: done.

File: iocage_test.yml
Comment (vbotka) [reply iocage_test.yml]: done

**Comment by vbotka** [comment]
> I'm happy with the diff , will try to test it soon on my box and then kick it to `master`.
> 
> Thank you for your work!!

I thank you! It's a bit messy. I've fixed the f-strings in a new path instead of editing the already submitted one. This is the subject of the pending "Changes request", I think.

**Comment by vbotka** [comment]
We have a problem. I didn't realize *ansible-test* is testing sanity also with Python 2.7 and 2.6. The same error reports 3.5

> ERROR: Found 1 compile issue(s) on python 2.7 which need to be resolved:
ERROR: plugins/modules/system/iocage.py:230:102: SyntaxError: module.fail_json(msg=f"{label}\ncmd: '{cmd}' return: {rc}\nstdout: '{stdout}'\nstderr: '{stderr}'")

**Comment by xmj** [comment]
> We have a problem. f-strings are available since 3.6 I didn't realize _ansible-test_ is testing sanity also with Python 2.7 and 2.6. The same error reports 3.5
> 
> > shell> ansible-test sanity plugins/modules/system/iocage.py --docker -v
> > ERROR: Found 1 compile issue(s) on python 2.7 which need to be resolved:
> > ERROR: plugins/modules/system/iocage.py:230:102: SyntaxError: module.fail_json(msg=f"{label}\ncmd: '{cmd}' return: {rc}\nstdout: '{stdout}'\nstderr: '{stderr}'")

`Python 2.7` is way past its [end of life](https://www.python.org/doc/sunset-python-2/) per the official Python Software Foundation timeline. Let's not worry about it.

**Comment by vbotka** [comment]
> > We have a problem. f-strings are available since 3.6 I didn't realize _ansible-test_ is testing sanity also with Python 2.7 and 2.6. The same error reports 3.5
> > > shell> ansible-test sanity plugins/modules/system/iocage.py --docker -v
> > > ERROR: Found 1 compile issue(s) on python 2.7 which need to be resolved:
> > > ERROR: plugins/modules/system/iocage.py:230:102: SyntaxError: module.fail_json(msg=f"{label}\ncmd: '{cmd}' return: {rc}\nstdout: '{stdout}'\nstderr: '{stderr}'")
> 
> `Python 2.7` is way past its [end of life](https://www.python.org/doc/sunset-python-2/) per the official Python Software Foundation timeline. Let's not worry about it.

Right. This means, ``Let's not worry about ansible-test`` (with respect to supported versions of Python). In practice, this means we won't be able to put the module into any Ansible collection. That said, our decision is explicit ``the module will live here and be installed on-demand``. I'm fine with it.

I'll put a note into the docs about Python supported from 3.6. I think the module should also test sanity and exit gracefully when Python < 3.6

**Comment by xmj** [comment]
> > > We have a problem. f-strings are available since 3.6 I didn't realize _ansible-test_ is testing sanity also with Python 2.7 and 2.6. The same error reports 3.5
> > > > shell> ansible-test sanity plugins/modules/system/iocage.py --docker -v
> > > > ERROR: Found 1 compile issue(s) on python 2.7 which need to be resolved:
> > > > ERROR: plugins/modules/system/iocage.py:230:102: SyntaxError: module.fail_json(msg=f"{label}\ncmd: '{cmd}' return: {rc}\nstdout: '{stdout}'\nstderr: '{stderr}'")
> > 
> > 
> > `Python 2.7` is way past its [end of life](https://www.python.org/doc/sunset-python-2/) per the official Python Software Foundation timeline. Let's not worry about it.
> 
> Right. This means, `Let's not worry about ansible-test` (with respect to supported versions of Python). In practice, this means we won't be able to put the module into any Ansible collection. That said, our decision is explicit `the module will live here and be installed on-demand`. I'm fine with it.
> 
> I'll put a note into the docs about Python supported from 3.6. I think the module should also test sanity and exit gracefully when Python < 3.6

Sounds great, thank you very much.

Notes for the future:
1. F-Strings were introduced in [PEP-498, which was adopted in Python 3.6](https://www.python.org/dev/peps/pep-0498/)
2. The [default python version on FreeBSD is 3.8](https://cgit.freebsd.org/ports/tree/Mk/bsd.default-versions.mk#n109)
3. The Expiration Date for `Python 2.7` on FreeBSD was [December 31, 2021](https://cgit.freebsd.org/ports/tree/lang/python27/Makefile#n16). Expect the port removed and support dropped "any time now" ;-)

**Review by xmj — COMMENTED** [review]
Changes look good to me, will merge after testing.

**Review by xmj — APPROVED** [review]

**Comment by xmj** [comment]
Merged after liberal use of `git rebase` (using fixup & squash) as #29

**Pull request closed without merging**
